### PR TITLE
MAINT: use copy=False in a few astype() calls

### PR DESCRIPTION
**Pull request by argriffing**
This is like https://github.com/scikit-learn/scikit-learn/issues/4573. I wasn't able to see any quantifiable speed or memory improvements.

### CODE REVIEW

File: numpy/linalg/linalg.py [hunk 1]
```diff
@@ -1749,7 +1750,8 @@ def det(a):
     _assertNdSquareness(a)
     t, result_t = _commonType(a)
     signature = 'D->D' if isComplexType(t) else 'd->d'
-    return _umath_linalg.det(a, signature=signature).astype(result_t)
+    r = _umath_linalg.det(a, signature=signature)
+    return r.astype(result_t)
```
Comment (jaimefrio) [inline numpy/linalg/linalg.py]: Did you miss the `copy=False` here?

Comment (argriffing) [inline numpy/linalg/linalg.py]: There is no `copy=False` for `astype` of a numpy scalar, I think.
```python
>>> import numpy as np
>>> np.float64(42).astype(float)
42.0
>>> np.float64(42).astype(float, copy=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: astype() takes no keyword arguments
```

Comment (jaimefrio) [inline numpy/linalg/linalg.py]: Ah, I see... There also seem to be tests that would catch if someone tried to introduce that optimization in the future, so I guess it is OK.

Perhaps it is worth a comment? What about branching on `np.isscalar(r)`?

**Comment by argriffing** [comment]
> Perhaps it is worth a comment? What about branching on `np.isscalar(r)`?

OK I've added this in the most recent commit.

**Comment by jaimefrio** [comment]
Merged, thanks!

**Pull request merged**